npm/webpack-dev-server/src/makeWebpackConfig.ts

```diff
@@ -1,28 +1,51 @@
 import { debug as debugFn } from 'debug'
 import * as path from 'path'
-import { Configuration } from 'webpack'
+import * as webpack from 'webpack'
 import { merge } from 'webpack-merge'
+import defaultWebpackConfig from './webpack.config'
+import LazyCompilePlugin from 'lazy-compile-webpack-plugin'
 import CypressCTOptionsPlugin, { CypressCTOptionsPluginOptions } from './plugin'
 
 const debug = debugFn('cypress:webpack-dev-server:makeWebpackConfig')
+const WEBPACK_MAJOR_VERSION = Number(webpack.version.split('.')[0])
+
+export interface UserWebpackDevServerOptions {
+  /**
+   * if `true` will compile all the specs together when the first one is request and can slow up initial build time.
+   * @default false
+  */
+  disableLazyCompilation?: boolean
+}
+
+interface MakeWebpackConfigOptions extends CypressCTOptionsPluginOptions, UserWebpackDevServerOptions {
+  webpackDevServerPublicPathRoute: string
+  isOpenMode: boolean
+}
 
 const mergePublicPath = (baseValue, userValue = '/') => {
   return path.join(baseValue, userValue, '/')
 }
 
-interface MakeWebpackConfigOptions extends CypressCTOptionsPluginOptions {
-  webpackDevServerPublicPathRoute: string
+function getLazyCompilationWebpackConfig (options: MakeWebpackConfigOptions): webpack.Configuration {
+  if (options.disableLazyCompilation || !options.isOpenMode) {
+    return {}
+  }
+
+  switch (WEBPACK_MAJOR_VERSION) {
+    case 4:
+      return { plugins: [new LazyCompilePlugin()] }
+    case 5:
+      return { experiments: { lazyCompilation: true } } as webpack.Configuration
```

Comment:
Right - so webpack 5 has lazy built in, but webpack 4 you need the plugin?

Reply:
I think this should be from the code <3